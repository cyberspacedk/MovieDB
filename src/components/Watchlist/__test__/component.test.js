import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import WatchList from '../component';

describe('Component: WatchList', () => {
  const props = {
    watchList: [{ id: 1, title: 'Some title', overview: 'Some description' }],
    loading: false,
    error: false,
    empty: false,
    currentPage: 1,
    totalResults: 25,
    history: {
      push: () => {},
    },
    operationsWatchListRequest: () => {},
    goToNextPage: jest.fn(),
  };

  it('Match its snapshot. Array provided', () => {
    const wrapper = shallow(
      <WatchList {...props} title="WatchList" removeBox />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Match its snapshot. Loading ...', () => {
    const nextProps = {
      ...props,
      favorites: [],
      loading: true,
    };
    const wrapper = shallow(
      <WatchList {...nextProps} title="Favorites" removeBox />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Match its snapshot. Error', () => {
    const nextProps = {
      ...props,
      watchList: [],
      error: true,
    };
    const wrapper = shallow(
      <WatchList {...nextProps} title="Favorites" removeBox />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('Match its snapshot. Empty', () => {
    const nextProps = {
      ...props,
      watchlist: [],
      empty: true,
    };
    const wrapper = shallow(
      <WatchList {...nextProps} title="Favorites" removeBox />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
