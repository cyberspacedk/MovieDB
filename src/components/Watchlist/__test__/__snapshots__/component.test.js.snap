// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: WatchList Match its snapshot. Array provided 1`] = `
<PageLayout
  array={
    Array [
      Object {
        "id": 1,
        "overview": "Some description",
        "title": "Some title",
      },
    ]
  }
  currentPage={1}
  empty={false}
  error={false}
  goToNextPage={[MockFunction]}
  history={
    Object {
      "push": [Function],
    }
  }
  loading={false}
  operations={[Function]}
  removeBox={true}
  title="Watchlist"
  totalResults={25}
/>
`;
